lib/session.js

```diff
@@ -117,13 +117,17 @@ function Session(conn) {
   this.mapped = false;
   this.remoteChannel = undefined;
   this._allocatedHandles = {};
-  this._linksByName = {};
-  this._linksByRemoteHandle = {};
   this._deliveryTag = 1;
 
   this._senderLinks = {};
   this._receiverLinks = {};
 
+  this._linksByName = {};
```

Comment:
Since we're only using these to resolve incoming attach frames, perhaps we can rename it something like _attachPendingLinks and `delete` the entry when we receive the attach. Right now it seems like we're double-book-keeping with `_senderLinks`/`_receiverLinks`.